packages/components/molecules/f-skeleton-loader/src/components/SkeletonLoader.vue

```diff
@@ -0,0 +1,75 @@
+
+
+<script>
+export default {
+    name: 'SkeletonLoader',
+    components: {},
+    props: {
+        skeletonType: {
+            type: String,
+            default: 'heading'
+        }
+    },
+    data () {
+        return {
+            rootTypes: {
+                text: 'text',
+                heading: 'heading'
+            }
+        };
+    },
+    methods: {
+        genBone (text = this.skeletonType, children) {
+            return this.$createElement(
+                'div',
+                {
+                    staticClass: `${this.$style[`c-skeleton-loader__${text}`]} ${this.$style['c-skeleton-loader__bone']}`
+                },
+                children
+            );
+        }
+    },
+    render (createElement) {
+        return createElement(
+            'div', {
+                staticClass: this.$style['c-skeleton-loader'],
+                attrs: {
+                    'data-test-id': 'skeletonLoader'
+                }
+            },
+            [this.genBone()]
+        );
+    }
+
+};
+</script>
+
+<style lang="scss" module>
+
+$skeleton-loader-heading-height: 32px;
+$skeleton-loader-text-height: 16px;
+
+.c-skeleton-loader {
+  position: relative;
+  vertical-align: top;
+
+   &__bone {
```

Comment:
We should be following [our CSS naming conventions](https://fozzie.just-eat.com/documentation/css/css-naming) here, i.e. children are denoted by a single dash, and modifiers by a double-dash so, in this case, the class would be `.c-skeleton-loader-bone`.

We also advise against using the ampersand shorthand as it makes finding classes more difficult, for example, say you had the "c-skeleton-loader-bone" class on an element and wanted to find it in code you wouldn't find it by searching for "c-skeleton-loader-bone". There are [some docs which mention this here](https://fozzie.just-eat.com/documentation/css/sass).

Finally, using our naming convention, these classes no longer need to be nested within the `.c-skeleton-loader` class.

Hope that all makes sense, give me a shout if you have any questions 😺

Reply:
@DamianMullins thanks I have not written CSS in a long time so this is very useful.

Reply:
No worries, CSS looks perfect now 😺